src/components/map/rent-popup.tsx

```diff
@@ -74,10 +78,14 @@ const RentControls: React.FC<RentControlsProps> = ({
 
         <button
           className="btn outline book"
-          disabled={!canBookBike}
-          onClick={onBookBike}
+          disabled={!canRentBike && !booking}
+          onClick={!booking ? onBookBike : onCancelBooking}
         >
-          {MAP.POPUP.BUTTON.BOOK}
+          {!booking ?
+            map.BOOKING.BOOK_BIKE :
+              BUCHUNGEN.RESERVIERUNG.BUTTON +
+              (!isCurrentStationBooked ? ` (${bookedStation && bookedStation.name})` : '')
```

Comment:
I'd rather use a template literal string for the whole thing here and not only for the last part. So `${BUCHUNGEN.RESERVIERUNG.BUTTON} ${...ternary}`.

Reply:
A template literal on a single line:
```suggestion
              `${BUCHUNGEN.RESERVIERUNG.BUTTON} ${!isCurrentStationBooked ? bookedStation!.name : ''}
```
(shortened thanks to the non-null assertion operator)

Reply:
See b833bedc802bc138f548cf6886285bcb0cf942e7.